components/policy-groups/PolicySearchAndResults.js

```diff
@@ -0,0 +1,166 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import Loading from "components/Loading";
+import styles from "styles/modules/PolicyGroupAssignments.module.scss";
+import { usePaginatedFetch } from "hooks/usePaginatedFetch";
+import Button from "components/Button";
+import { stateActions } from "reducers/appState";
+import {
+  DEFAULT_DEBOUNCE_DELAY,
+  PLAYGROUND_SEARCH_PAGE_SIZE,
+  SEARCH_ALL,
+} from "utils/constants";
+import PolicySearchBar from "components/policies/PolicySearchBar";
+import { useAppState } from "providers/appState";
+import useDebouncedValue from "hooks/useDebouncedValue";
+import { createSearchFilter } from "utils/shared-utils";
+import Icon from "components/Icon";
+import { ICON_NAMES } from "utils/icon-utils";
+
+const PolicySearchAndResults = ({ onAssign, assignedToGroup }) => {
+  const [policySearch, setPolicySearch] = useState(false);
+
+  const [debounceDelay, setDebounceDelay] = useState(DEFAULT_DEBOUNCE_DELAY);
+  const { state, dispatch } = useAppState();
+  const debouncedSearch = useDebouncedValue(
+    state.policySearchTerm,
+    debounceDelay
+  );
+
+  const { data, loading, isLastPage, goToNextPage } = usePaginatedFetch(
+    debouncedSearch ? "/api/policies" : null,
+    createSearchFilter(debouncedSearch),
+    PLAYGROUND_SEARCH_PAGE_SIZE
+  );
+
+  const assignedPolicyIds = assignedToGroup.map((assignment) => {
+    const [policyId] = assignment.policyVersionId.split(".");
+    return policyId;
+  });
+
+  return (
+    <div className={styles.searchContainer}>
+      <PolicySearchBar
+        onSubmit={(event) => {
+          event.preventDefault();
+          if (!state.policySearchTerm.length) {
+            dispatch({
+              type: stateActions.SET_POLICY_SEARCH_TERM,
+              data: SEARCH_ALL,
+            });
+          }
+          setPolicySearch(true);
+        }}
+        onBlur={() => setDebounceDelay(0)}
+        onChange={() => {
+          setDebounceDelay(DEFAULT_DEBOUNCE_DELAY);
+        }}
+        helpText={
+          <Button
+            buttonType={"text"}
+            label={"View all policies"}
+            onClick={() => {
+              dispatch({
+                type: stateActions.SET_POLICY_SEARCH_TERM,
+                data: SEARCH_ALL,
+              });
+              setPolicySearch(true);
+            }}
+            className={styles.viewAllButton}
+          />
+        }
+      />
+      <div className={styles.searchResultsContainer}>
+        {policySearch && (
+          <Loading loading={loading} type={"button"}>
+            {data?.length > 0 ? (
+              <>
+                {data.map((result) => {
+                  const isSelected = assignedPolicyIds.includes(result.id);
```

Comment:
Maybe slightly less efficient, but could this check against the version ids instead?:

```javascript
const assignedPolicyIds = assignedToGroup.map((assignment) => assignment.policyVersionId.split("."));

const isSelected = assignedPolicyIds.some((versionId) => versionId.includes(result.id));
```

I think we should avoid depending on the structure of the policy version id too much, it's only the way that it is because we can't constrain uniqueness in another way.